wandb/sdk/launch/sweeps/scheduler.py

```diff
@@ -377,40 +408,28 @@ def _update_run_states(self) -> None:
 
         Get state from backend and deletes runs if not in running state. Threadsafe.
         """
-        # TODO(gst): move to better constants place
-        end_states = [
-            "crashed",
-            "failed",
-            "killed",
-            "finished",
-            "preempted",
-        ]
-        run_states = ["running", "pending", "preempting"]
-
-        _runs_to_remove: List[str] = []
+        runs_to_remove: List[str] = []
         for run_id, run in self._yield_runs():
             try:
-                _state = self._api.get_run_state(self._entity, self._project, run_id)
-                _rqi_state = run.queued_run.state if run.queued_run else None
-                if not _state or _state in end_states or _rqi_state == "failed":
-                    _logger.debug(
-                        f"({run_id}) run-state:{_state}, rqi-state:{_rqi_state}"
-                    )
-                    run.state = RunState.DEAD
-                    _runs_to_remove.append(run_id)
-                elif _state in run_states:
-                    run.state = RunState.ALIVE
+                state = self._api.get_run_state(self._entity, self._project, run_id)
+                run.state = RunState(state)
             except CommError as e:
-                _logger.debug(
-                    f"Issue when getting state for run ({run_id}) with error: {e}"
-                )
+                _logger.debug(f"error getting state for run ({run_id}): {e}")
+                if run.state == RunState.UNKNOWN:
+                    # triggers when we get an unknown state for the second time
+                    wandb.termwarn(f"Failed to get runstate for run ({run_id}")
```

Comment:
Set the run to failed, so it gets cleaned up and killed. Propagate error message to user, (include stacktrace?)